sklearn/neighbors/tests/test_neighbors_tree.py

```diff
@@ -294,3 +294,14 @@ def test_pickle(Cls, metric, protocol):
     assert_array_almost_equal(dist1, dist2)
 
     assert isinstance(tree2, Cls)
+
+
+@pytest.mark.parametrize("BinaryTree", (BallTree, KDTree))
+def test_ball_tree_valid_metrics(BinaryTree):
+    """BinaryTrees must expose a public `valid_metrics` class attribute containing
+    distance metric string identifiers.
+
+    Non-regression test for #26742"""
+
+    valid_metrics_list = BinaryTree.valid_metrics
+    assert all(isinstance(valid_metric, str) for valid_metric in valid_metrics_list)
```

Comment:
I don't think we need such a test. This attribute is already tested by other code, tests and doctest.

And the fact that it is public API is already stated by the fact that there is no leading underscore in the attribute name, the class name and the module name, and the fact that it is documented in `doc/modules/neighbors.rst`.